pkg/trace/traceutil/processed_trace.go

```diff
@@ -18,3 +18,36 @@ type ProcessedTrace struct {
 	TracerHostname         string
 	ClientDroppedP0sWeight float64
 }
+
+// Clone creates a copy of ProcessedTrace, cloning p, p.TraceChunk, and p.Root. This means it is
```

Comment:
Thanks for this detailed godoc. SUPER helpful in keeping the code safe going forward